### DISCUSSION TITLE
JWT가 무엇이고, 세션 대신 사용하는 이유가 무엇인가요?

### DISCUSSION THREAD

**Discussion by alstn113** [2024-woowacourse-study/level-interview#164] [Spring 뒷고기]
https://velog.io/@cis07385/%EB%A9%B4%EC%A0%91-%EC%98%88%EC%83%81-%EC%A7%88%EB%AC%B83

**Comment by gyungchan-jo** [comment]
JWT는 JSON Web Token의 약자로, 웹에서 사용되는 토큰 기반의 인증 방식 중 하나입니다. JWT는 주로 사용자 인증을 위해 사용되며, 세 가지 부분으로 구성됩니다. 각각의 부분들은 Base64Url로 인코딩되어 있으며, 각각 "."으로 구분되어 있습니다.

- Header: JWT 토큰의 타입과 알고리즘 정보를 포함하는 JSON 객체입니다. 보통 JWT를 인코딩할 때 Base64Url로 인코딩되어 헤더 파트에 추가됩니다.
- Payload: JWT에 포함될 정보를 담고 있는 JSON 객체입니다. 클라이언트가 서버로 전달하려는 정보, 사용자 식별자, 권한 등이 포함될 수 있습니다. 마찬가지로 Base64Url로 인코딩되어 페이로드 파트에 추가됩니다.
- Signature: JWT를 인증하기 위한 서명 정보입니다. 서명을 생성하기 위해서는 Header와 Payload, 그리고 사전에 정해진 비밀키가 필요합니다. 보통 Header와 Payload를 합쳐서 하나의 문자열로 만든 후, 비밀키를 이용하여 서명합니다. 서명은 Base64Url로 인코딩되어 토큰 뒷부분에 추가됩니다.

---

세션 대신 사용하는 이유:

세션은 인증이나 인가를 하는 과정에서 서버 자원에 접근을 합니다. 이유는 사용자의 데이터가 담긴 데이터는 서버 측에서 관리를 하기 때문입니다.
반면, jwt는 사용자의 정보가 토큰의 payload에 저장되어 있기 때문에 서버 자원에 접근하지 않더라도 인증이나 인가를 할 수 있어 서버의 자원을 사용하는 횟수가 적어 세션 대신 JWT를 사용합니다.
